src/logic/useReading.tsx

```diff
@@ -27,8 +28,12 @@ export const useReading = () => {
 // Provider hook that creates auth object and handles state
 function useProvideReading() {
   const [reading, setReading] = useState(InitialReading);
-  // const dummyGenerator = DummyDataGenerator(setReading, DataConfig.dataFrequency);
-  const serialDataHandler = SerialDataHandler({ baudRate: 115200 }, setReading);
+  const dataGateWay = (data: any) => {
+    setReading(data);
+    dataLeecher(data);
+  }
+  const dummyGenerator = DummyDataGenerator(dataGateWay, DataConfig.dataFrequency);
+  const serialDataHandler = SerialDataHandler({ baudRate: 115200 }, dataGateWay);
```

Comment:
should we not be passing in dataLeecher, which will have an update function which is called, and that then calls all its subscribed functions?

Reply:
yes